### DISCUSSION TITLE
Option to highlight specific region in PlotBrowserTrack

### DISCUSSION THREAD

**Discussion by AnjaliC4** [GreenleafLab/ArchR#1342] [Feature Requests]
Hi @rcorces, 

I think it would be a very helpful addition to highlight a region of interest in the PlotBrowserTrack function (to highlight a region overlapping with a SNP for example)- similar to Gviz functionality. For ArchR, this can be done by adding a geom_rect() function to ggplot when plotting genomic regions. I don't have the expertise to add this to PlotBrowserTrack source code, but if you can point me to a line of code where this can be added - I can do it. 

Basically, hoping for something like this: https://stackoverflow.com/questions/32543176/highlight-areas-within-certain-x-range-in-ggplot2

Thanks

**Comment by rcorces** [comment]
This is a nice suggestion. You would need to edit each of the individual plotting functions so that the highlighting appears in each portion of the browser track (each bulk track, the gene track, coaccessibility track, peak2gene track, etc).
This requires editing `.bulkTracks()` `.scTracks()` `.featureTracks()` `.loopTracks()` and `.geneTracks()` and editing `plotBrowserTrack()` to include a new parameter (likely a `GRanges` object) containing the regions to be highlighted.

I can try to look into this but it will take time.

**Reply by NoemieL** [reply]
Hi Ryan,

Did you had a chance to get this parameter available?

Thanks

**Reply by rcorces** [reply]
not yet. Happy to review a pull request if someone wants to implement it.

**Reply by paupaiz** [reply]
@rcorces please assign this to me

**Reply by rcorces** [reply]
@paupaiz - please take this on! I cant assign discussions to individuals (only Issues).

**Comment by AnjaliC4** [comment]
Hi @NoemieL & @rcorces, 

I have already created a working code for this. You have to add a valid genomic ranges object as the region you wish to highlight. In the function, you can do  p <- plotBrowserTrack(proj, highlight=GRobjToHighlight) 

https://github.com/AnjaliC4/ArchR_plots/blob/main/Highlight_ArchRTracks.R

Let me know if you have any questions or find any issues with this. 

Anjali

**Reply by rcorces** [reply]
thanks @AnjaliC4 
That link doesnt work for me though. Maybe the repo is private.

**Reply by NoemieL** [reply]
Thanks @AnjaliC4 to share this. However as Ryan mentioned it is not accessible. Could you please put your repo as public? Thank you so much

**Reply by AnjaliC4** [reply]
Hi @rcorces  and @NoemieL,

I have made my repo public. You should now be able to access the code. You will have to source(Highlight_ArchRTracks.R) and call the function normally: https://github.com/AnjaliC4/ArchR_plots/blob/main/Highlight_ArchRTracks.R

P.S. I can also create a pull-request if you'd prefer @rcorces 

Let me know how it goes. Cheers!

**Reply by NoemieL** [reply]
Thank you very much @AnjaliC4! I works well!

**Reply by rcorces** [reply]
@AnjaliC4 @NoemieL @paupaiz - I had some time today and implemented this on the branch `dev_highlightTrack`. I changed a bit of how it was done and made it robust to some obvious user input errors. I also added an option to change the color of the highlighted region.

```
#' @param highlight A `GRanges` object containing a region on the plot to highlight. Multiple highlighted regions within the GRanges object are allowed
#' Any highlight region that does not overlap the displayed region will be ignored.
#' @param highlightColor The color to be used for the highlighted region designated by `highlight`. This can be a valid R color (i.e. "lightblue1")
#' or a hex color (i.e. "#bfefff")
```

Please let me know if this works for you or if you encounter bugs. If it works, I will merge into `release_1.0.2`

**Reply by rcorces** [reply]
Ultimately, this was merged into the current `dev` branch and will become part of `release_1.0.3`
